File: module-services/src/main/java/eu/geekhome/services/configurable/RequiredStringValidator.java
package eu.geekhome.services.configurable;


import eu.geekhome.services.localization.Resource;

public class RequiredStringValidator implements Validator<String> {
    @Override
    public Resource getReason() {
        return new Resource("This field is required",
                "To pole jest wymagane");
    }

    @Override
    public boolean validate(String fieldValue) {
        return fieldValue != null && !fieldValue.isEmpty();
    }
}

File: module-services/src/main/java/eu/geekhome/services/configurable/RequiredStringValidator.kt
package eu.geekhome.services.configurable

import eu.geekhome.services.R
import eu.geekhome.services.localization.Resource

class RequiredStringValidator : Validator<String> {

    override fun validate(fieldValue: String): Boolean {
        return fieldValue.isNotEmpty()
    }

    override val reason: Resource = R.validator_required_field
}